### DISCUSSION TITLE
`maxBounds` LateInitialisationError Bug

### DISCUSSION THREAD

**Discussion by WenYiCoding** [fleaflet/flutter_map#1328] [Q&A]
This is the bug I found when adding `maxBounds` attribute to restrict the map scrolling.

Let's say:
`maxBounds:
          LatLngBounds(
             LatLng(3.219, 101.711),   // A, B
             LatLng(3.200, 101.700),  // a, b
),`
The difference between `A` and `a` only can be small as 0.019.
Meanwhile, the difference between `B` and `b` only can as small as 0.011.

This means if I decrease the value of A,B or increase the value of a,b I will get this error.
![image](https://user-images.githubusercontent.com/107612053/182163049-0cc49687-71da-4432-9c43-fbe018f5f20d.png)

I hope the maintenance team can investigate the problem.

**Comment by ibrierley** [comment]
I suspect this isn't a precision bug, but assuming this is the same issue as in Discord, more that there are an incompatible settings, that result in no available options to satisfy the conditions.

Your maxBounds param tries to force bounds that are inside your bounds param. So it can't satisfy this and barfs. If you remove your bounds parameter you will probably find it works fine.

**Reply by JaffaKetchup** [reply]
I would probably agree with this, I was just taking a quick guess over in Discord.

**Comment by jetpeter** [comment]
@WenYiCoding try out the code in https://github.com/fleaflet/flutter_map/pull/1331.  I would be interested to know if it resolves your issue.

If you run into an issue where the map is drawn with a border outside your maxBounds and you cannot move it, make sure you set your maxZoom to a value high enough to fit the bounds on your screen.
